src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java

```diff
@@ -127,14 +127,23 @@ private String createFileForAuxClasspath(SpotBugsTask task) {
           Paths.get(
               task.getProject().getBuildDir().getAbsolutePath(),
               "spotbugs",
-              "spotbugs-auxclasspath");
-      Files.createDirectories(auxClasspathFile.getParent());
-      Files.createFile(auxClasspathFile);
-      Files.write(auxClasspathFile, auxClasspath.getBytes(), StandardOpenOption.TRUNCATE_EXISTING);
-      return auxClasspathFile.normalize().toString();
+              "spotbugs-auxclasspath-for-" + task.getName());
+      try {
+        Files.createDirectories(auxClasspathFile.getParent());
+        if (Files.exists(auxClasspathFile)) {
+          Files.delete(auxClasspathFile);
+        }
+        Files.createFile(auxClasspathFile);
+        Files.write(auxClasspathFile, auxClasspath.getBytes(), StandardOpenOption.WRITE);
```

Comment:
What is our motivation to replace `TRUNCATE_EXISTING` with `WRITE`?
and I guess `TRUNCATE_EXISTING` lets us remove the lines 133-136? (sorry I didn't noticed it previously)

Reply:
The motivation was to delete and create the file each time, but the outcome should be taken care of with the directory hierarchy however.